docs/upgrading.rst

```diff
@@ -0,0 +1,156 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2015-2019 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+Upgrading to version 3.2
+========================
+
+If you have your instance of Invenio v3.1 already up and running and
+you would like to upgrade to version v3.2 you don't need to set up your
+project from scratch. The goal of this guide is to show the steps to upgrade
+your project without losing any of your work.
+
+
+Pipfile modifications
+---------------------
+
+The most important changes that you will have to make are in ``Pipfile``.
+
+First you need to change the invenio version:
+
+::
+
+    invenio = { version = "==3.2.0a7", extras = ["base", "auth", "metadata", "files", "postgresql", "elasticsearch7" ]}
+
+If you want to use the new files bundle make sure you include the ``files``
+bundle. Add any additional :ref:`bundles` you would like in your project in
+``extras``.
+
+To install the new packages in ``Pipfile`` run the following commands:
+
+.. code:: bash
+
+    # Update Pipfile.lock
+    pipenv lock --dev
+
+    # Install packages specified in Pipfile.lock
+    pipenv sync --dev
+
+    # Install application code and entrypoints from 'setup.py'
+    pipenv run pip install -e .
+
+    # Build assets
+    pipenv run invenio collect -v
+    pipenv run invenio webpack buildall
+
+
+Database
+--------
+Changes have been made to the database from Invenio 3.1 so you will need to
+upgrade the database by running the latest Alembic recipes:
+
+.. code:: bash
+
+    invenio alembic upgrade
+
+Your database should now have the latest changes.
+
+
+Files
+-----
+To integrate the files bundle with your Invenio instance, please see the guide
+to configure files for Invenio 3.2.
+
+For files to work properly ensure that the config variables
+``RECORDS_FILES_REST_ENDPOINTS`` and ``FILES_REST_PERMISSION_FACTORY`` have been
+configured properly.
+
+.. note::
+
+    If you are upgrading from a previous cookiecutter instance and you updated
+    ``records/config.py``, please remember to update the changed config keys in
+    ``records/ext.py``.
+
+Uploading files
+~~~~~~~~~~~~~~~
+Records created after you upgraded to Invenio 3.2 will support files
+out-of-the-box as long as files are configured properly.
+
+However, if you have records created by previous versions of Invenio they will
+not work out-of-the-box with files because there is no bucket attached to
```

Comment:
```suggestion
they require to apply changes in record metadata
```
if you mention `bucket` here, please link it with the documentation as people might not know what the bucket is at this point